### ISSUE TITLE
omprog: add ability to re-use rsyslog's environment

### ISSUE THREAD

**Issue by rgerhards** [rsyslog/rsyslog#2921]
right now (8.37), rsyslog provides an empty environment to the spawned program. This seems to work in existing use case. However, there are use cases where rsyslog's environment should be inherited, at least this is needed for the testbench. There, we need to have access to the output file name, which is now dynamically be generated to support parallel test execution. I would assume environment access would also be useful for other use cases as well.

@jsiwrk  @davidelang do you think this is actually an enhancement or was it a bug in previous versions? This decides if we need a config switch for it and what the default value is.

**Comment by jsiwrk** [comment]
I can't think of a realistic use case where passing an empty environment to the child can be considered a feature (security considerations don't seem to apply here, since the program is anyway being executed with the same privileges as rsyslog). So adding the config switch wouldn't be necessary IMO. However, it's also OK to add it, as I see you have already done. So OK for my part.

**Comment by rgerhards** [comment]
I can undo the config switch if that is consensus. I just added it to be
able to carry on with my Testbench work. Not a big deal to just fix the
issue if we agree it's a bug and change of behavior desirable (I also tend
to this POV).

Rainer

Sent from phone, thus brief.

**Comment by davidelang** [comment]
On Mon, 13 Aug 2018, Rainer Gerhards wrote:

> I can undo the config switch if that is consensus. I just added it to be
> able to carry on with my Testbench work. Not a big deal to just fix the
> issue if we agree it's a bug and change of behavior desirable (I also tend
> to this POV).

I agree.

David Lang

**Comment by rgerhards** [comment]
Yes, I will remove. Sorry, I did not see someone waited for this. Happens
most probably tomorrow.

Rainer

Sent from phone, thus brief.

Joan Sala <notifications@github.com> schrieb am Do., 16. Aug. 2018, 02:20:
